src/network/gcs_routes.cpp

```diff
@@ -260,7 +303,7 @@ DEF_GCS_HANDLE(Post, takeoff, manual) {
     LOG_REQUEST("POST", "takeoff/manual");
 
     auto lock_ptr = state->getTickLockPtr<WaitForTakeoffTick>();
-    if (!lock_ptr.has_value()) {
+    if (!lock_ptr.has_value() || state->getTickID() != TickID::WaitForTakeoff) {
```

Comment:
this can be removed and then this pr should be ready to merge (I fixed the tick lock ptr so that it this isn't needed, since the has value call should catch if it is the wrong tick)

Reply:
Done